src/core/Header/HeaderButton.tsx

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+import {
+  Button,
+  ButtonProps,
+  DropdownButton,
+  DropdownButtonProps,
+} from '../Buttons';
+
+import { useTheme } from '../utils/hooks/useTheme';
+
+export type HeaderButtonProps = {
+  /**
+   * Name.
+   */
+  name: React.ReactNode;
+  /**
+   * Description
```

Comment:
Looks really poor :D
Could you add maybe some more wording? description could have mentioned the font size change for example

Reply:
I'm not aware of a font-size change, but it does disappear in slim mode.